### DISCUSSION TITLE
Middleware matcher config for Next Auth

### DISCUSSION THREAD

**Discussion by mertafor** [amannn/next-intl#292] [Q&A]
Apparently this library is using config to skip routes that will not be translated. 

```
export const config = {
  // Skip all paths that should not be internationalized
  matcher: ['/((?!api|_next|.*\\..*).*)']
};
```

However, Next Auth library is also using matcher to secure certain routes. Apparently if I define matcher for the routes that should be secured by next auth, those routes will also be excluded from next-intl translations. Is there any way to set matcher for different cases like this?

By the way this is probably a NextJS middleware question rather than next-intl, but I'd appreciate any help.

**Comment by amannn** [comment]
Hey @mertafor! Have you seen the docs on [NextAuth.js](https://next-intl-docs.vercel.app/docs/next-13/middleware#example-auth-js)? There are some hints in regard to this and there's also a link to a working example.

Hope this helps!

**Reply by mertafor** [reply]
Docs didn't help because advanced middleware solution requires withAuth wrapper. But later I found some solution on twitter. Instead of using config variable, getting pathname from request, comparing with an array and using a custom middleware  is the answer. 

Thanks for your reply and great library!

**Reply by amannn** [reply]
Interesting! Can you share your configuration in case other developers run into the same situation in the future and stumble over this thread?

**Reply by mertafor** [reply]
@amannn I've just done that :)

**Comment by mertafor** [comment]
In case it helps someone : 

```
import createIntlMiddleware from 'next-intl/middleware'
import { NextRequest, NextResponse } from 'next/server'
import { getToken } from 'next-auth/jwt'
import { LANGUAGES } from './constants/globalCons'

// auth paths
const requireAuth: string[] = ['/admin', '/dashboard']

export default async function middleware(request: NextRequest) {
  const defaultLocale = request.headers.get('x-default-locale') || 'tr'

  const pathname = request.nextUrl.pathname
  if (requireAuth.some((path) => pathname.startsWith(path))) {
    const token = await getToken({
      req: request,
      secret: process.env.SECRET
    })
    if (!token) {
      const url = new URL(`/login`, request.url)
      url.searchParams.set('callbackUrl', encodeURI(request.url))
      return NextResponse.redirect(url)
    }
    //check if not authorized
    /* if (pathname.startsWith('/admin') && token.role !== "admin") {
        const url = new URL(`/403`, request.url);
        return NextResponse.rewrite(url);
      } */
  }

  const handleI18nRouting = createIntlMiddleware({
    locales: LANGUAGES,
    defaultLocale
  })
  const response = handleI18nRouting(request)

  response.headers.set('x-default-locale', defaultLocale)

  return response
}

export const config = {
  matcher: ['/((?!api|_next|.*\\..*).*)']
}
```
